src/pages/steps/EventTypeAndThemeStep.tsx

```diff
@@ -144,11 +212,34 @@ const EventTypeAndThemeStep = ({
     scope,
   });
 
-  const types = getTypesByScopeQuery.data ?? [];
+  const types = useMemo(
+    () => getTypesByScopeQuery.data ?? [],
+    [getTypesByScopeQuery.data],
+  );
```

Comment:
If I remove the useMemo here I'll get a warning by the hooks eslint plugin:

```
The 'types' logical expression could make the dependencies of useMemo Hook (at line 221) change on every render
```

The `themes` contant is dependent on the `types` constant. Since `types` can technically be a new array every time (and will re-trigger a re-render every time) it's safer to wrap it in a `useMemo`.

Reply:
Ah yes I see I thought the hook returned a direct useQuery result and that as such it would only change once after initial load but looking into the hook there's more happening there so makes sense 👍

Reply:
Yeah I think the eslint rule is mainly complaining about  `?? []`. Ad long as `getTypesByScopeQuery.data` is `undefined`, a new array reference will be set to `types`.